wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/interfaces/TermedStatementDocument.java

```diff
@@ -0,0 +1,53 @@
+package org.wikidata.wdtk.datamodel.interfaces;
+
+/*
+ * #%L
+ * Wikidata Toolkit Data Model
+ * %%
+ * Copyright (C) 2014 - 2018 Wikidata Toolkit Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This interface just joins {@link TermedDocument} and {@link StatementDocument}.
+ * 
+ * It is necessary to introduce this interface because the conflict between 
+ * the return types of the withRevisionId method in both interfaces needs to be resolved.
+ * @author antonin
+ *
+ */
+public interface TermedStatementDocument extends TermedDocument, StatementDocument {
+
+	@Override
+	TermedStatementDocument withRevisionId(long newRevisionId);
+	
+	@Override
+	TermedStatementDocument withLabel(MonolingualTextValue newLabel);
+	
+	@Override
+	TermedStatementDocument withDescription(MonolingualTextValue newDescription);
+	
+	@Override
+	TermedStatementDocument withAliases(String language, List<MonolingualTextValue> aliases);
```

Comment:
Is there a good reason to have the language code as first argument even if it is already contained in `MonolingualTextValue`?
What about just a simple `TermedStatementDocument withAlias(MonolingualTextValue alias);` method?

Reply:
The thing is that this *replaces* all aliases in the given language by the list supplied. This gives more control, because it lets people remove all aliases by passing the empty list (hence the language parameter is needed).

Reply:
I see. What about having more explicit methods like `withoutAliases(String language)` that removes all aliases in a language and `withAlias(MonolingualTextValue alias)` to add an alias (and maybe `withoutAlias(MonolingualTextValue alias)` to remove an alias?

Reply:
With these methods you still cannot control the order of the aliases, which is stored by wikibase…

Reply:
Good point. But it is not significant on the semantic level so I hope nobody cares. But I'm ok with your current method if you are afraid of this problem.